text/0149-exponential-histogram.md

```diff
@@ -0,0 +1,162 @@
+
+```
+scale   #subBuckets base         relative_error
+-3       8          1.090507733  4.43%
+-4      16          1.044273782  2.19%
+-5      32          1.021897149  1.09%
+```
+
+The [comment in #226](https://github.com/open-telemetry/opentelemetry-proto/pull/226#issuecomment-777922339) has more details on why 2 was chosen as the reference base. In summary, "computers like binary numbers". If we are going to choose a reference base, why not make it 2 ("10" in binary)?
+
+### Base10 vs. base2
+
+Alternatively, we may choose a reference base of 10 (decimal). While this may have some "human friendliness", in practice, the benefit is minimal. As shown in the table below, in base10, baseScale of interest is around -6, where a log10 bucket is sub-divided into 64 sub-buckets. We get a bound like 10, 100, or 1000 only every 64 buckets. The "human friendliness" value is minimal.
+
+```
+scale   #subBuckets base         relative error
+-5       32         1.074607828  3.66%
+-6       64         1.036632928  1.82%
+-7      128         1.018151722  0.90%
+```
+
+Let's further consider the following use cases of histograms to compare base2 and base10:
+
+1. Displaying histogram charts. With a typical base around 1.04 (around 2% relative error), there will be hundreds of buckets for a typical range over 100x. This many points can produce a reasonably smooth curve, regardless of the raw data being base2 or base10.
+2. Calculating percentiles or quantiles. This is often used in SLO monitoring. Example SLO: "99% percentile of response time need to be no more than 100ms". To minimize relative error, percentile calculation usually returns log scale mid point of a bucket. So returned percentile values won't be on 10, 100, 1000, etc., even if the histogram is base10.
+3. Answering question like "what percentage of values fall below 100". When the threshold is on 10, 100, 1000, etc, base10 histograms do give exact answer. But even in the decimal world, power of 10 numbers is a small population. For thresholds like 200, 500, etc. base10 histograms have no advantage over base2. If an exact answer is required in these cases, a user should create explicit buckets at these bounds, instead of using exponential buckets.
+
+So the "human friendliness" of base10 exponential histograms is largely an illusion. To some extent, the base2 vs. base10 question has been answered long ago: computers convert input from base10 to base2, do processing in base2, then convert final output from base2 to base10. In the histogram case, we take input in "double", which is already in [binary float point format](https://en.wikipedia.org/wiki/Double-precision_floating-point_format). Bucketing these numbers in base10 bounds effectively switches base during processing. It just adds complexity and computational cost.
```

Comment:
I agree with the conclusion in the context of this discussion: For log-histograms there is little point in using base-10.

I see a real usability advantage of using base-10 when using log-linear bins (...,100,110,...200,210,...,1000,1100,..) since these cover all the interesting cases for humans with regards to question 3. ("what percentage of values fall below 100"). I think this is the point that others have been making in related discussions. For the purpose of the current scope (log-bins) the benefits with regards to (3) are marginal.